*** include/io.h.orig	Fri Nov 19 13:13:44 2010
--- include/io.h	Mon Nov 22 09:00:46 2010
***************
*** 375,381 ****
--- 375,383 ----
  
  /* SH_... flags for nShFlags defined in share.h
   * Optional fourth argument is unsigned unPermissions */
+ #ifdef WINVER
  _CRTIMP int __cdecl __MINGW_NOTHROW _sopen (const char*, int, int, ...);
+ #endif
  
  _CRTIMP long __cdecl __MINGW_NOTHROW _tell (int);
  /* Should umask be in sys/stat.h and/or sys/types.h instead? */
***************
*** 451,457 ****
--- 453,461 ----
  _CRTIMP int __cdecl __MINGW_NOTHROW open (const char*, int, ...);
  _CRTIMP int __cdecl __MINGW_NOTHROW read (int, void*, unsigned int);
  _CRTIMP int __cdecl __MINGW_NOTHROW setmode (int, int);
+ #ifdef WINVER
  _CRTIMP int __cdecl __MINGW_NOTHROW sopen (const char*, int, int, ...);
+ #endif
  _CRTIMP long __cdecl __MINGW_NOTHROW tell (int);
  _CRTIMP int __cdecl __MINGW_NOTHROW umask (int);
  _CRTIMP int __cdecl __MINGW_NOTHROW unlink (const char*);
